magellan-library/src/main/java/com/wealthfront/magellan/lifecycle/LifecycleStateMachine.kt

```diff
@@ -33,7 +33,7 @@ internal class LifecycleStateMachine {
       currentState = when (currentState.getTheDirectionIShouldGoToGetTo(newState)) {
         FORWARD -> next(subjects, currentState, newState.context!!)
         BACKWARDS -> previous(subjects, currentState, getContext(newState, oldState))
-        NO_MOVEMENT -> throw IllegalStateException("Attempting to transition from $currentState to itself")
+        NO_MOVEMENT -> newState
```

Comment:
When does this happen? Wouldn't transitioning to the current state mean there's a problem somewhere?

Reply:
Yeah. Im not sure about this. Im gonna add some logging maybe. But removed it since it was not harmful.

Reply:
Could we keep it until it causes problems? I'd rather fail fast than have this in an unexpected state, personally

Reply:
Oh, i think i figured it out. We previously were comparing the instances of the data classes in the while loop. Switched it over to compare the order instead. I think that should resolve the issue. Keeping the throw statement.